SQL null value error in migration after relative path fix

Got a new SQL error in migration :

Application fails to boot.

2022-01-29 09:38:27.772  WARN --- ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'themeSource': Unsatisfied dependency expressed through method 'setSettingsService' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'settingsService': Unsatisfied dependency expressed through field 'internetRadioDao'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-media-file-id-to-user-rating::anon:
     Reason: liquibase.exception.DatabaseException: ERROR: null value in column "media_file_id" of relation "user_rating" violates not-null constraint
  Detail: Failing row contains (rune, /mnt/nas/Music/Musikk/Kenny Rogers/(kenny rogers) the gambler.mp..., 5, null). [Failed SQL: (0) UPDATE public.user_rating SET media_file_id = (SELECT media_file.id FROM media_file WHERE media_file.path=user_rating.path)]
2022-01-29 09:38:27.813 ERROR --- o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'themeSource': Unsatisfied dependency expressed through method 'setSettingsService' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'settingsService': Unsatisfied dependency expressed through field 'internetRadioDao'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-media-file-id-to-user-rating::anon:
     Reason: liquibase.exception.DatabaseException: ERROR: null value in column "media_file_id" of relation "user_rating" violates not-null constraint
  Detail: Failing row contains (rune, /mnt/nas/Music/Musikk/Kenny Rogers/(kenny rogers) the gambler.mp..., 5, null). [Failed SQL: (0) UPDATE public.user_rating SET media_file_id = (SELECT media_file.id FROM media_file WHERE media_file.path=user_rating.path)]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.resolveMethodArguments(AutowiredAnnotationBeanPostProcessor.java:768) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.inject(AutowiredAnnotationBeanPostProcessor.java:720) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1413) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:213) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1160) ~[spring-context-5.3.8.jar:5.3.8]
        at org.springframework.ui.context.support.UiApplicationContextUtils.initThemeSource(UiApplicationContextUtils.java:58) ~[spring-context-5.3.8.jar:5.3.8]
        at org.springframework.web.context.support.GenericWebApplicationContext.onRefresh(GenericWebApplicationContext.java:183) ~[spring-web-5.3.8.jar:5.3.8]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:158) ~[spring-boot-2.5.2.jar:2.5.2]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:577) ~[spring-context-5.3.8.jar:5.3.8]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145) ~[spring-boot-2.5.2.jar:2.5.2]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-2.5.2.jar:2.5.2]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:434) ~[spring-boot-2.5.2.jar:2.5.2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:338) ~[spring-boot-2.5.2.jar:2.5.2]
        at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:144) ~[spring-boot-2.5.2.jar:2.5.2]
        at org.airsonic.player.Application.main(Application.java:70) ~[classes/:na]
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'settingsService': Unsatisfied dependency expressed through field 'internetRadioDao'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-media-file-id-to-user-rating::anon:
     Reason: liquibase.exception.DatabaseException: ERROR: null value in column "media_file_id" of relation "user_rating" violates not-null constraint
  Detail: Failing row contains (rune, /mnt/nas/Music/Musikk/Kenny Rogers/(kenny rogers) the gambler.mp..., 5, null). [Failed SQL: (0) UPDATE public.user_rating SET media_file_id = (SELECT media_file.id FROM media_file WHERE media_file.path=user_rating.path)]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:660) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:640) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1413) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.resolveMethodArguments(AutowiredAnnotationBeanPostProcessor.java:760) ~[spring-beans-5.3.8.jar:5.3.8]
        ... 21 common frames omitted
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-media-file-id-to-user-rating::anon:
     Reason: liquibase.exception.DatabaseException: ERROR: null value in column "media_file_id" of relation "user_rating" violates not-null constraint
  Detail: Failing row contains (rune, /mnt/nas/Music/Musikk/Kenny Rogers/(kenny rogers) the gambler.mp..., 5, null). [Failed SQL: (0) UPDATE public.user_rating SET media_file_id = (SELECT media_file.id FROM media_file WHERE media_file.path=user_rating.path)]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:660) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:640) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1413) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:657) ~[spring-beans-5.3.8.jar:5.3.8]
        ... 35 common frames omitted
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-media-file-id-to-user-rating::anon:
     Reason: liquibase.exception.DatabaseException: ERROR: null value in column "media_file_id" of relation "user_rating" violates not-null constraint
  Detail: Failing row contains (rune, /mnt/nas/Music/Musikk/Kenny Rogers/(kenny rogers) the gambler.mp..., 5, null). [Failed SQL: (0) UPDATE public.user_rating SET media_file_id = (SELECT media_file.id FROM media_file WHERE media_file.path=user_rating.path)]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1786) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:602) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:657) ~[spring-beans-5.3.8.jar:5.3.8]
        ... 49 common frames omitted
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-media-file-id-to-user-rating::anon:
     Reason: liquibase.exception.DatabaseException: ERROR: null value in column "media_file_id" of relation "user_rating" violates not-null constraint
  Detail: Failing row contains (rune, /mnt/nas/Music/Musikk/Kenny Rogers/(kenny rogers) the gambler.mp..., 5, null). [Failed SQL: (0) UPDATE public.user_rating SET media_file_id = (SELECT media_file.id FROM media_file WHERE media_file.path=user_rating.path)]
        at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:124) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.lambda$null$0(Liquibase.java:265) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.lambda$update$1(Liquibase.java:264) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.runInScope(Liquibase.java:2404) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.update(Liquibase.java:211) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.Liquibase.update(Liquibase.java:197) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:314) ~[liquibase-core-4.6.2.jar:na]
        at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:269) ~[liquibase-core-4.6.2.jar:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845) ~[spring-beans-5.3.8.jar:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782) ~[spring-beans-5.3.8.jar:5.3.8]
        ... 61 common frames omitted

Similar error here:

```
2022-01-29 11:59:40.329  WARN --- ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'themeSource': Unsatisfied dependency expressed through method 'setSettingsService' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'settingsService': Unsatisfied dependency expressed through field 'internetRadioDao'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
     Reason: liquibase.exception.DatabaseException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)] [Failed SQL: (-104) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)]
2022-01-29 11:59:40.672 ERROR --- o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'themeSource': Unsatisfied dependency expressed through method 'setSettingsService' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'settingsService': Unsatisfied dependency expressed through field 'internetRadioDao'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
     Reason: liquibase.exception.DatabaseException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)] [Failed SQL: (-104) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.resolveMethodArguments(AutowiredAnnotationBeanPostProcessor.java:768) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.inject(AutowiredAnnotationBeanPostProcessor.java:720) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1413) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:213) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1160) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.ui.context.support.UiApplicationContextUtils.initThemeSource(UiApplicationContextUtils.java:58) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.web.context.support.GenericWebApplicationContext.onRefresh(GenericWebApplicationContext.java:183) ~[spring-web-5.3.8.jar:5.3.8]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:158) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:577) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:434) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:338) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:144) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.airsonic.player.Application.main(Application.java:70) ~[classes/:na]
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'settingsService': Unsatisfied dependency expressed through field 'internetRadioDao'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
     Reason: liquibase.exception.DatabaseException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)] [Failed SQL: (-104) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:660) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:640) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1413) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.resolveMethodArguments(AutowiredAnnotationBeanPostProcessor.java:760) ~[spring-beans-5.3.8.jar:5.3.8]
	... 21 common frames omitted
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
     Reason: liquibase.exception.DatabaseException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)] [Failed SQL: (-104) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:660) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:640) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1413) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:657) ~[spring-beans-5.3.8.jar:5.3.8]
	... 35 common frames omitted
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
     Reason: liquibase.exception.DatabaseException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)] [Failed SQL: (-104) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1786) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:602) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:657) ~[spring-beans-5.3.8.jar:5.3.8]
	... 49 common frames omitted
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
     Reason: liquibase.exception.DatabaseException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)] [Failed SQL: (-104) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)]
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:124) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Liquibase.lambda$null$0(Liquibase.java:265) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Liquibase.lambda$update$1(Liquibase.java:264) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Liquibase.runInScope(Liquibase.java:2404) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Liquibase.update(Liquibase.java:211) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Liquibase.update(Liquibase.java:197) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:314) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:269) ~[liquibase-core-4.6.2.jar:na]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845) ~[spring-beans-5.3.8.jar:5.3.8]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782) ~[spring-beans-5.3.8.jar:5.3.8]
	... 61 common frames omitted
Caused by: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
     Reason: liquibase.exception.DatabaseException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)] [Failed SQL: (-104) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)]
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:695) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:49) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:111) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:110) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:94) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.Scope.child(Scope.java:244) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:66) ~[liquibase-core-4.6.2.jar:na]
	... 79 common frames omitted
Caused by: liquibase.exception.DatabaseException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)] [Failed SQL: (-104) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)]
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:397) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:83) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:151) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1279) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1261) ~[liquibase-core-4.6.2.jar:na]
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:660) ~[liquibase-core-4.6.2.jar:na]
	... 99 common frames omitted
Caused by: java.sql.SQLIntegrityConstraintViolationException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ur_u_mfid_pk PRIMARY KEY (username, media_file_id)]
	at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.jdbc.JDBCStatement.fetchResult(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.jdbc.JDBCStatement.execute(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:94) ~[HikariCP-4.0.3.jar:na]
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java) ~[HikariCP-4.0.3.jar:na]
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:393) ~[liquibase-core-4.6.2.jar:na]
	... 104 common frames omitted
Caused by: org.hsqldb.HsqlException: integrity constraint violation: unique constraint or index violation; UR_U_MFID_PK table: USER_RATING
	at org.hsqldb.error.Error.error(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.Constraint.getException(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.index.IndexAVL.insert(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.persist.RowStoreAVL.indexRow(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.persist.RowStoreAVL.moveData(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.TableWorks.moveData(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.TableWorks.addPrimaryKey(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.StatementSchema.getResult(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.StatementSchema.execute(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.Session.executeCompiledStatement(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.Session.executeDirectStatement(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.Session.execute(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	... 109 common frames omitted
```

--- a/airsonic-main/src/main/resources/liquibase/11.0/relative-media-file-paths.xml
+++ b/airsonic-main/src/main/resources/liquibase/11.0/relative-media-file-paths.xml
@@ -110,6 +110,15 @@
             </addColumn>
         </rollback>
     </changeSet>
+    <changeSet id="clean-user-rating-for-absent-paths" author="anon">
+        <preConditions onFail="MARK_RAN">
+            <columnExists tableName="user_rating" columnName="path"/>
+        </preConditions>
+        <delete tableName="user_rating">
+            <where>path IN (SELECT ur.path FROM user_rating ur LEFT JOIN media_file m ON ur.path=m.path WHERE m.path IS NULL)</where>
+        </delete>
+        <rollback></rollback>
+    </changeSet>
     <changeSet id="add-media-file-id-to-user-rating" author="anon">
         <preConditions onFail="MARK_RAN">
             <not>
@@ -121,10 +130,32 @@
                 <constraints nullable="false" />
             </column>
         </addColumn>
+        <rollback>
+            <dropColumn tableName="user_rating" columnName="media_file_id"></dropColumn>
+        </rollback>
+    </changeSet>
+    <changeSet id="populate-user-rating-media-file-id" author="anon">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <sqlCheck expectedResult="0">select count(*) from user_rating where media_file_id = '-1'</sqlCheck>
+            </not>
+        </preConditions>
+        <update tableName="user_rating">
+            <column name="media_file_id" valueComputed="(SELECT media_file.id FROM media_file WHERE media_file.path=user_rating.path)"></column>
+            <where>media_file_id='-1'</where>
+        </update>
+        <rollback>
+        </rollback>
+    </changeSet>
+    <changeSet id="add-user-rating-media-file-id-fk" author="anon">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <foreignKeyConstraintExists foreignKeyTableName="user_rating" foreignKeyName="ura_m_fk"/>
+            </not>
+        </preConditions>
         <addForeignKeyConstraint constraintName="ura_m_fk" baseTableName="user_rating" baseColumnNames="media_file_id" referencedTableName="media_file" referencedColumnNames="id" onDelete="CASCADE"/>
         <rollback>
             <dropForeignKeyConstraint baseTableName="user_rating" constraintName="ura_m_fk"/>
-            <dropColumn tableName="user_rating" columnName="media_file_id"></dropColumn>
         </rollback>
     </changeSet>
     <changeSet id="drop-user-rating-path-pk" author="anon">
@@ -153,6 +184,15 @@
             </addColumn>
         </rollback>
     </changeSet>
+    <changeSet id="clean-user-rating-for-uniqueness" author="anon">
+        <preConditions onFail="WARN">
+            <columnExists tableName="user_rating" columnName="media_file_id"/>
+        </preConditions>
+        <delete tableName="user_rating">
+            <where>media_file_id = '-1'</where>
+        </delete>
+        <rollback></rollback>
+    </changeSet>
     <changeSet id="add-user-rating-pk" author="anon">
         <preConditions onFail="MARK_RAN">
             <columnExists tableName="user_rating" columnName="media_file_id"/>
@@ -167,6 +207,15 @@
             <dropForeignKeyConstraint baseTableName="user_rating" constraintName="ura_u_fk"/>
             <dropPrimaryKey tableName="user_rating" constraintName="ur_u_mfid_pk"/>
         </rollback>
+    </changeSet>
+    <changeSet id="clean-share-file-for-absent-paths" author="anon">
+        <preConditions onFail="MARK_RAN">
+            <columnExists tableName="share_file" columnName="path"/>
+        </preConditions>
+        <delete tableName="share_file">
+            <where>path IN (SELECT sf.path FROM share_file sf LEFT JOIN media_file m ON sf.path=m.path WHERE m.path IS NULL)</where>
+        </delete>
+        <rollback></rollback>
     </changeSet>
     <changeSet id="add-media-file-id-to-share-file" author="anon">
         <preConditions onFail="MARK_RAN">

--- a/airsonic-main/src/main/resources/liquibase/11.0/relative-media-file-paths.xml
+++ b/airsonic-main/src/main/resources/liquibase/11.0/relative-media-file-paths.xml
@@ -115,7 +115,7 @@
             <columnExists tableName="user_rating" columnName="path"/>
         </preConditions>
         <delete tableName="user_rating">
-            <where>path IN (SELECT ur.path FROM user_rating ur LEFT JOIN media_file m ON ur.path=m.path WHERE m.path IS NULL)</where>
+            <where>path IN (SELECT * FROM (SELECT ur.path FROM user_rating ur LEFT JOIN media_file m ON ur.path=m.path WHERE m.path IS NULL) tbl_tmp)</where>
         </delete>
         <rollback></rollback>
     </changeSet>
@@ -213,7 +213,7 @@
             <columnExists tableName="share_file" columnName="path"/>
         </preConditions>
         <delete tableName="share_file">
-            <where>path IN (SELECT sf.path FROM share_file sf LEFT JOIN media_file m ON sf.path=m.path WHERE m.path IS NULL)</where>
+            <where>path IN (SELECT * FROM (SELECT sf.path FROM share_file sf LEFT JOIN media_file m ON sf.path=m.path WHERE m.path IS NULL) tbl_tmp)</where>
         </delete>
         <rollback></rollback>
     </changeSet>

